### ISSUE TITLE
ImprovedCameraAE

### ISSUE THREAD

**Issue by CyrusWalker** [tiltedphoques/Mod-Compatibility#9]
# Mod information

#### Download link
https://github.com/ArranzCNL/ImprovedCameraAE/releases/tag/v1.0.0

#### Dependencies

- SKSE 2.1.5
  -  http://skse.silverlock.org/

# Compatibility

> Mod is compatible with Pre Release 2 of ImprovedCameraAE

# Additional Information

> Thank you ArranzCNL for making these mods compatible!

**Comment by ArranzCNL** [comment]
I see that is coming from Improved Camera's keyboard patch no doubt the mouse will need patching as well.

I'll give this mod a spin later fixing these startup errors, no doubt there are more!

**Comment by CyrusWalker** [comment]
> I see that is coming from Improved Camera's keyboard patch no doubt the mouse will need patching as well.
> 
> I'll give this mod a spin later fixing these startup errors, no doubt there are more!

You're a champ! Love me some Improved Camera :D

**Comment by ArranzCNL** [comment]
Ok seemed to just be the keyboard that needed a patch on my end to get this to work.

I have issues with our DInput8 hook failing which needs looking into but this can be disabled anyway HookInput=0 and would mean to use the overlay MenuMode=1, this mode cannot be ran in pure fullscreen though, borderless fullscreen is fine.

I did notice a conflict with STR, don't open IC's menu if STR has a menu open otherwise STR breaks and your only option is to ALT+F4 the game.

You do have to alter the ini file as they change the window name to Skyrim Together:

> ; Application window name. If you use Skyrim Together remove ; and place a ; infront of the other.
> WindowName=Skyrim Special Edition
> ;WindowName=Skyrim Together

I'll alter the ini on next release to above, hopefully it doesn't sound confusing.

[Teaser Pic](https://imgur.com/a/7KL68Cu)

**Comment by CyrusWalker** [comment]
> Ok seemed to just be the keyboard that needed a patch on my end to get this to work.
> 
> I have issues with our DInput8 hook failing which needs looking into but this can be disabled anyway HookInput=0 and would mean to use the overlay MenuMode=1, this mode cannot be ran in pure fullscreen though, borderless fullscreen is fine.
> 
> I did notice a conflict with STR, don't open IC's menu if STR has a menu open otherwise STR breaks and your only option is to ALT+F4 the game.
> 
> You do have to alter the ini file as they change the window name to Skyrim Together:
> 
> > ; Application window name. If you use Skyrim Together remove ; and place a ; infront of the other.
> > WindowName=Skyrim Special Edition
> > ;WindowName=Skyrim Together
> 
> I'll alter the ini on next release to above, hopefully it doesn't sound confusing.
> 
> [Teaser Pic](https://imgur.com/a/7KL68Cu)

So just change the ini to have: HookInput=0, MenuMode=1 or 0, and WindowName = Skyrim Together ?

**Comment by ArranzCNL** [comment]
> So just change the ini to have: HookInput=0, MenuMode=1 or 0, and WindowName = Skyrim Together ?

Yep that is correct, Improved Camera AE PR2 should be out this week at some point.

**Comment by CyrusWalker** [comment]
> > So just change the ini to have: HookInput=0, MenuMode=1 or 0, and WindowName = Skyrim Together ?
> 
> Yep that is correct, Improved Camera AE PR2 should be out this week at some point.

Tried this fix and I am still having the same problem. I'll wait for the next official release to see if any of your more in-depth changes work.

**Comment by ArranzCNL** [comment]
Sorry maybe I was not clear, your not going to get around the keyboard issue until next update. The rest of what I mentioned shall be how to get it working with one another.

**Comment by CyrusWalker** [comment]
> Sorry maybe I was not clear, your not going to get around the keyboard issue until next update. The rest of what I mentioned shall be how to get it working with one another.

Ah okay, thanks for the clarification. I will await your next update patiently :)

**Comment by ArranzCNL** [comment]
Just to give you a final update on this, end users don't need to do anything now due to a mod detection system in place in the upcoming build. Assume Vortex works the same way if not well they'll have to do it manually, as I only officially support MO2.

**Comment by ArranzCNL** [comment]
[Improved Camera AE Preview Release 2](https://github.com/ArranzCNL/ImprovedCameraAE/releases) is now available and is compatible.

**Comment by CyrusWalker** [comment]
> [Improved Camera AE Preview Release 2](https://github.com/ArranzCNL/ImprovedCameraAE/releases) is now available and is compatible.

You're amazing!

**Comment by Pingu-Pox** [comment]
@CyrusWalker You should update your initial post and update the tags on the post.

**Comment by CyrusWalker** [comment]
> 

I can update the post, but not the tags, that one is on one of the devs to take care of.

**Comment by Dragonissa** [comment]
> @CyrusWalker You should update your initial post and update the tags on the post.

![image](https://user-images.githubusercontent.com/4233773/180644236-c1a62214-47ef-4a07-9f0f-51f9999b5781.png)

**Comment by jmanginger98** [comment]
The links to this mod return a 404 error.

**Comment by PirateWithoutACause** [comment]
Latest Update: https://www.nexusmods.com/skyrimspecialedition/mods/93962?tab=files

**Comment by SunTrustDev** [comment]
When using the latest version I get 500% startup time on the client itself.
e.g. from 30-40 seconds to 3 minutes+
Am I the only?

**Comment by Andromeda3953** [comment]
@SunTrustDev 
> When using the latest version I get 500% startup time on the client itself. e.g. from 30-40 seconds to 3 minutes+ Am I the only?

I know im super late to this but, to fix this long start up time, go into ImprovedCameraSE.INI and change Application Window Name to "Skyrim Together", and CheckCompatibility to "1" (no quotes on both ofc)

**Comment by GemsKnight** [comment]
Hello!
With STR+Community Shader+Improved Camera SE, I had a CTD on client start. I managed to make it work by resetting the CheckCompatibility to 0. Something in the loading was crashing the client, it allows it to skip it and all seem to work fine for now

**Comment by DiligentPickle7** [comment]
> > When using the latest version I get 500% startup time on the client itself. e.g. from 30-40 seconds to 3 minutes+ Am I the only?
> 
> I know im super late to this but, to fix this long start up time, go into ImprovedCameraSE.INI and change Application Window Name to "Skyrim Together", and CheckCompatibility to "1" (no quotes on both ofc)

So I was getting some weird fps capping with both those settings enabled and ended up only using the Application Window Name change and forgoing the CheckCompatibility, which somehow fixed it.

Idk why or how this happened, but if your fps is getting capped to like 48 for some reason, try this
